rust_src/src/lisp.rs

```diff
@@ -130,13 +141,31 @@ pub enum LispType {
     Lisp_Float = 7,
 }
 
+impl LispObject {
+    #[allow(unused_unsafe)]
+    pub unsafe fn get_type_unchecked(self) -> LispType {
+        let res = (self.to_raw() & !VALMASK) as u8;
+        unsafe { mem::transmute(res) }
+    }
+
+    #[inline]
+    pub unsafe fn get_untaggedptr_unchecked(self) -> * mut libc::c_void {
```

Comment:
It is safe as long as every number in usize is a valid pointer. In casual cases it's always true. I'm not sure if there's corner cases. I'm remove the unsafe anyway.